src/Model/Map.php

```diff
@@ -2,15 +2,216 @@
 
 namespace CommonsBooking\Model;
 
+use CommonsBooking\Repository\Item;
+use CommonsBooking\Repository\Timeframe;
+use CommonsBooking\Wordpress\CustomPostType\Location;
+use Exception;
+
 /**
- * This class currently does not have any functionality.
- * It's just a placeholder for the Map post type.
- * Because the map feature was taken from another plugin, it is not yet adapted to the structure of this plugin.
- * This class will be used to adapt the map feature to the structure of this plugin.
+ * This class does the heavy lifting for the map shortcode
+ * Code style differs because it has been taken from the fLotte Map shortcode plugin
  *
- * Currently, it's only referenced by the @see \CommonsBooking\Wordpress\CustomPostType\CustomPostType::getModel() method to use the methods from the CustomPost class.
  */
-class Map extends CustomPost
-{
+class Map extends CustomPost {
+
+	/**
+	 * get geo data from location metadata
```

Comment:
Also hier werden eigentlich die Geodaten inkl. Meta-Daten, welche nötig sind für die Anzeige der Kartenrepräsentation geladen?

Reply:
Basierend auf den Map-Settings.

Reply:
Das war früher in der CustomPostType Klasse, ich hab das nur umgezogen, weil ich es für Logik halte, die besser in das Model passt. Eine Model-Klasse gab es früher noch nicht. Die Karte ist ja aus der alten 0.9.X Kartenerweiterung übernommen worden. Deshalb folgt da auch nix unseren Standards.

Reply:
Verstehe. Mir ging es oben auch eher um den recht spärlichen PHPdoc, wenn die Funktion ja einige recht verborgene Dinge tut (Terms, Timeframes, Map-Meta-Felder, Location-Meta-Felder) ide aus "relevant" oder "metadata" jetzt nicht abzuleiten sind.